anta/tests/path_selection.py

```diff
@@ -0,0 +1,148 @@
+                    elif not session:
+                        self.result.is_failure(f"Telemetry state for peer {peer} in group {group} is `inactive`.")
+
+
+class VerifySpecificRouterPath(AntaTest):
+    """
+    Verifies the route and telemetry state of a specific path for an IPv4 peer under router path-selection.
+
+    The expected states are 'IPsec established', 'Resolved' for route and 'active' for telemetry.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the path under router path-selection has its route state as either 'IPsec established' or 'Resolved'
+               and telemetry state as 'active'.
+    * Failure: The test will fail if router path-selection is not configured, the path's route state is not 'IPsec established' or 'Resolved',
+               or the telemetry state is 'inactive'.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.path_selection:
+      - VerifySpecificRouterPath:
+          paths:
+            - peer: 10.255.0.1
+              path_groups:
+                - internet
+                - mpls
+    ```
+    """
+
+    name = "VerifySpecificRouterPath"
+    description = "Verifies the route and telemetry state of a specific path under router path-selection."
+    categories: ClassVar[list[str]] = ["path-selection"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaTemplate(template="show path-selection paths peer {peer} path-group {group}")]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifySpecificRouterPath test."""
+
+        paths: list[RouterPath]
+        """List of router paths to verify."""
+
+        class RouterPath(BaseModel):
+            """Detail of a router path."""
+
+            peer: IPv4Address
+            """Static peer IPv4 address."""
+
+            path_groups: list[str]
+            """Router path group names."""
+
+    def render(self, template: AntaTemplate) -> list[AntaCommand]:
+        """Render the template for each router path."""
+        return [template.render(peer=path.peer, group=path_group) for path in self.inputs.paths for path_group in path.path_groups]
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifySpecificRouterPath."""
+        self.result.is_success()
+
+        # Check the state of each path
+        for command in self.instance_commands:
+            peer = command.params.peer
+            path_group = command.params.group
+            command_output = command.json_output.get("dpsPeers", [])
+
+            # If the peer is not configured for the path group, the test fails
+            if not command_output:
+                self.result.is_failure(f"Peer `{peer}` is not configured for path group `{path_group}`.")
+                continue
+
+            # Extract the state of the path
+            path_output = get_value(command_output, f"{peer}..dpsGroups..{path_group}..dpsPaths", separator="..")
```

Comment:
wondering if you cannot get multiple Paths (it is a dict) for the same peer / path-group. And so potentiaally here, we woudl only be looking at the same.
I think it could happen if you have multiple local interface in the same path-group. Can you try this?